pandas/io/tests/test_json/test_pandas.py

```diff
@@ -324,12 +324,12 @@ def test_frame_to_json_except(self):
     def test_frame_empty(self):
         df = DataFrame(columns=['jim', 'joe'])
         self.assertFalse(df._is_mixed_type)
-        assert_frame_equal(read_json(df.to_json()), df)
+        assert_frame_equal(read_json(df.to_json()), df, check_dtype=False)
 
         # mixed type
         df['joe'] = df['joe'].astype('i8')
         self.assertTrue(df._is_mixed_type)
-        assert_frame_equal(read_json(df.to_json()), df)
+        assert_frame_equal(read_json(df.to_json()), df, check_dtype=False)
```

Comment:
dtypes need to be checked. Construct and expected frame that is correct and compare against it. round-tripping on json does not necessarily preserver dtypes. Note that you can also give dtype hints to the reader to make it match the original, so split this into 2 tests.